client/src/App.tsx

```diff
@@ -7,6 +7,7 @@ import { Menu } from './Navigation'
 import { PreferencesContext } from './Popups/Settings'
 import { useWindowDimensions } from './utils/window_width'
 import LeanLogo from './assets/logo.svg'
+import LZString from 'lz-string';
```

Comment:
This library has the MIT license.

I considered using [`CompressionStream`](https://developer.mozilla.org/en-US/docs/Web/API/CompressionStream) with gzip but it only became supported in Safari last year. I figured it would be bad if some live URLs (generated by people with modern browsers) didn't work on browsers that have a non-trivial amount of users left.

I came across lz-stream because it's used in [sharplab.io](https://sharplab.io), a C# REPL : https://github.com/ashmind/SharpLab/blob/078c4e2c5a790f6b7203401c2ec63de22913c0d1/source/WebApp/app/features/persistent-state/handlers/url.ts#L1

Reply:
Aside: sharplab does an extra thing it calls "pre-compression" where it replaces some common substrings with shorter ones (`@3`, `@7` etc.) before sending it to this library: https://github.com/ashmind/SharpLab/blob/main/source/WebApp/app/features/persistent-state/handlers/url/precompressor.ts

I did some quick checks and it wasn't a big deal. If you're interested I could get some actual data using the benchmark process I described in the PR description. The trade-off is extra complexity/.js size for a much smaller reduction in URL size.

Reply:
If you want to further optimise the algorithm, please feel free to do so, I'd happily merge it :)